docker/docker.go

```diff
@@ -727,3 +727,18 @@ func resizeTty(c *client.Client, containerID string) error {
 		Width:  width,
 	})
 }
+
+// NCPU returns the number of CPUs in the docker host
+func NCPU(ctx context.Context) (int, error) {
```

Comment:
could you please explain/give a link what number exactly this function returns. Quick search on the internet didn't answer my question. I'm not sure how it works on linux but on macOs (and most probably windows) number of available CPUs for docker is limited by default. On my laptop with 4 cores docker uses only 2 by default.
So docker can't freeze my system (only freeze itself, but I can kill it).

Does this function return real number of CPUs or only the number available to docker?

Reply:
Right, I forgot about docker desktop, that is already using a VM with limits.
On linux it returns the number of host CPUs, 8 in my case.

So I'm think that I can modify this to limit the container CPU only if the docker host is linux. Otherwise for mac or windows you have docker desktop, and it already comes with good default limits.

Reply:
I'm not aware of docker on linux in details. But if there is no default/manual limit for all docker then your solution sounds good.

Reply:
done in cb36df7